mautic/mautic
# 3 new contact batch actions

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/274
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/4843
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This is successor of https://github.com/mautic/mautic/pull/5858.

This PR adds new:
- **Category**
- **Channel subscription**
- **Owner**
batch options to the Contact List View.

#### Steps to test this PR:
1. Checkout this PR
2. Clear the cache
3. Go to the Contact list view.
4. Check some contacts
5. Click on the green button above the check fields and select one of the 3 options from the PR description.
6. Fill in some values to the modal form.
7. Check the contacts you've checked that the values are stored to them correctly. To do that go to the contact detail page, click the arrow button in the top right corner to open the contact's menu. Go the Preference Center and view the values there.

8. Since there were changes to the channel form, test the public side of it too:
    a. Make sure you have preference center switched ON in the Mautic configuration.
    b. Send a segment email to some contacts (yourself).
    c. Unsubscribe from some email you've got (use incognito browser) and ensure the contact has correct values pre-selected.
    d. Change the preferences, refresh the page and ensure your new values are still there.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
